app/src/pages/newPostPage/newPost.jsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from "react"
+import Footer from "../../components/footer/footer"
+import Header from "../../components/header/header"
+import ArtImage from "../../components/image/image"
+import styles from "./newPost.module.css"
+import Cookies from "universal-cookie"
+import { Navigate } from "react-router-dom"
+
+import API from '../../utils/api'
+
+
+export default function NewImage () {
+    const cookies = new Cookies(null, { path : "/"})
+    if (!cookies.get('images')){
+        cookies.set('images', [])
+    }
+
+    // only add draft images if they exist
+    const [images, setImages] = useState(cookies.get('images')? cookies.get('images') : [])
+    const [currentImage, setCurrentImage] = useState({"id" : null})
+    const [description, setDescription] = useState("")
+
+    const [selectedImages, setSelectedImages] = useState([])
+    const [navPending, setNavPending] = useState(false)
+
+    async function handleNewImage (e) {
+
+        if (!description) {return }
+
+        console.log(cookies.get("currentUser"))
+
+        const response = await API.postNewImage(cookies.get('currentUser').id, description)
+
+        if (response.status == 200){
+            let resJson = await response.json()
+            resJson['imagePrompt'] = description
+            resJson['user'] = cookies.get('currentUser')
+
+            // add generated images to current sessions generated images
```

Comment:
to clarify, generated images are stored in both cookie + firebase?

Reply:
Yes. 
The pipeline for a generated image is :
1. Receive temporary url from Ai, stored in cookie
2. Url in cookie is then submitted to firebase for permanent storage
3. Firebase Url and Id stored in Prisma